test/parallel/test-fs-open.js

```diff
@@ -32,8 +32,15 @@ fs.open(__filename, 'rs', function(err, fd) {
   openFd2 = fd;
 });
 
+assert.throws(function() {
+  fs.open('/path/to/file/that/does/not/exist', { bad: 'flags' }, assert.fail);
+}, function(err) {
+  if (err instanceof TypeError) {
```

Comment:
Well I wrote the error as a TypeError since its thrown on bad types only, so might as well test for it, no?